src/payment/strategies/stripev3/stripev3-payment-strategy.ts

```diff
@@ -330,8 +330,12 @@ export default class StripeV3PaymentStrategy implements PaymentStrategy {
 
         switch (element) {
             case StripePaymentMethodType.CreditCard:
-                const shippingAddress = this._store.getState().shippingAddress.getShippingAddress();
-                result = { ...result, shipping: this._mapStripeShippingAddress(shippingAddress, customer) };
+                const cart = this._store.getState().cart.getCart();
+
+                if (cart && cart.lineItems.physicalItems.length !== 0 ) {
```

Comment:
rather than checking the length of physical items, might be worth creating a generic function that takes the cart/lineItems and checks if `requiresShipping(cart): bool`

The logic should be:
```js
return cart.lineItems.physicalItems.filter(({ addedByPromotion }) => !addedByPromotion).length > 0
```

You could copy-paste this function:
https://github.com/bigcommerce/checkout-js/blob/master/src/app/shipping/getShippableItemsCount.ts